mypy/checkexpr.py

```diff
@@ -1136,26 +1136,62 @@ def visit_index_expr_helper(self, e: IndexExpr) -> Type:
             return result
 
     def visit_tuple_slice_helper(self, left_type: TupleType, slic: SliceExpr):
-        begin = 0
-        end = len(left_type.items)
-        stride = 1
+        begin = None  # type: int
+        end = None  # type: int
+        stride = None  # type:int
+
         if slic.begin_index:
+            ok = False
```

Comment:
Feel like I'm repeating myself for begin_index, end_index and stride. Wondering if it's worth moving the logic to an _function